### ISSUE TITLE
Store.best_justified_checkpoint can be updated with an older checkpoint

### ISSUE THREAD

**Issue by ericsson49** [ethereum/eth2.0-specs#1496]
Recent update to the [fork choice](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_fork-choice.md) spec introduced `best_justified_checkpoint` field in the [`Store`](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_fork-choice.md#store) class.
However, when it's updated in the [on_block](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_fork-choice.md#on_block) method, there is no check that the new value's epoch is newer than `store.best_justified_checkpoint`. The check only ensures that the new value's epoch is newer than the `store.justified_checkpoint`.
```
# Update justified checkpoint
if state.current_justified_checkpoint.epoch > store.justified_checkpoint.epoch:
    store.best_justified_checkpoint = state.current_justified_checkpoint
    if should_update_justified_checkpoint(store, state.current_justified_checkpoint):
        store.justified_checkpoint = state.current_justified_checkpoint
```

For example, if `store.justified_checkpoint.epoch == N` and there arrived a block which state has `current_justified_checkpoint == N+2`, `store.best_justified_checkpoint` becoming `N+2` and `current_justified_checkpoint` being still `N+2`, if the block arrived at the middle or at the end of `N+3` epoch.
If later another block arrives (still during the `N+3` epoch) which state has `current_justified_checkpoint == N+1`, then it will override the `N+2` value in the `store.best_justified_checkpoint`.

**Comment by djrtwo** [comment]
Nice catch!

fixing this now for the coming release

**Comment by djrtwo** [comment]
@ericsson49 A fix is in #1508 for review

**Comment by djrtwo** [comment]
closed via #1508
